controllers/nodemaintenance_controller_test.go

```diff
@@ -2,338 +2,367 @@ package controllers
-			Expect(maintenance.Status.Phase).NotTo(Equal(nodemaintenanceapi.MaintenanceRunning))
-			Expect(len(maintenance.Status.PendingPods)).NotTo(Equal(2))
-			Expect(maintenance.Status.EvictionPods).NotTo(Equal(2))
-			Expect(maintenance.Status.TotalPods).NotTo(Equal(2))
-			Expect(maintenance.Status.DrainProgress).To(Equal(0))
-			Expect(maintenance.Status.LastUpdate.IsZero()).To(BeTrue())
-		})
-
-	})
-
-	Context("Taint functioninality test", func() {
-		It("should add medik8s NoSchedule taint and keep other existing taints", func() {
-			node := &corev1.Node{}
-			err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: "node01"}, node)
-			Expect(err).NotTo(HaveOccurred())
-			_ = AddOrRemoveTaint(r.drainer.Client, node, true)
-			taintedNode := &corev1.Node{}
-			err = k8sClient.Get(context.TODO(), client.ObjectKey{Name: "node01"}, taintedNode)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(taintExist(taintedNode, "medik8s.io/drain", corev1.TaintEffectNoSchedule)).To(BeTrue())
-			Expect(taintExist(taintedNode, "node.kubernetes.io/unschedulable", corev1.TaintEffectNoSchedule)).To(BeTrue())
-			Expect(taintExist(taintedNode, "test", corev1.TaintEffectPreferNoSchedule)).To(BeTrue())
-
-			// there is a not-ready taint now as well... skip count tests
-			//Expect(len(taintedNode.Spec.Taints)).To(Equal(3))
-		})
-
-		It("should remove medik8s NoSchedule taint and keep other existing taints", func() {
-			node := &corev1.Node{}
-			err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: "node01"}, node)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(taintExist(node, "medik8s.io/drain", corev1.TaintEffectNoSchedule)).To(BeFalse())
-			_ = AddOrRemoveTaint(r.drainer.Client, node, true)
-			taintedNode := &corev1.Node{}
-			err = k8sClient.Get(context.TODO(), client.ObjectKey{Name: "node01"}, taintedNode)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(taintExist(taintedNode, "medik8s.io/drain", corev1.TaintEffectNoSchedule)).To(BeTrue())
-			_ = AddOrRemoveTaint(r.drainer.Client, taintedNode, false)
-			unTaintedNode := &corev1.Node{}
-			err = k8sClient.Get(context.TODO(), client.ObjectKey{Name: "node01"}, unTaintedNode)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(taintExist(unTaintedNode, "medik8s.io/drain", corev1.TaintEffectNoSchedule)).To(BeFalse())
-			Expect(taintExist(unTaintedNode, "test", corev1.TaintEffectPreferNoSchedule)).To(BeTrue())
-
-			//Expect(len(unTaintedNode.Spec.Taints)).To(Equal(1))
+		Context("Testing Taints", func() {
+			BeforeEach(func() {
+				Expect(k8sClient.Create(context.Background(), nodeOne)).To(Succeed())
+				DeferCleanup(k8sClient.Delete, context.Background(), nodeOne)
+			})
+			When("Adding new taint", func() {
+				It("should add new taint and keep other existing taints", func() {
+					node := &corev1.Node{}
+					Expect(k8sClient.Get(context.Background(), client.ObjectKey{Name: taintedNodeName}, node)).To(Succeed())
+					Expect(AddOrRemoveTaint(r.drainer.Client, node, true)).To(Succeed())
```

Comment:
Indeed. The whole taint usability and functions could be improved in NMO. I was thinking of moving the current [AppendTaint and RemoveTaint](https://github.com/medik8s/fence-agents-remediation/blob/main/pkg/utils/taints.go#L54-L111) that FAR and SNR uses to the common Medik8s API plus some minor changes. Then NMO could use it as well.
It might even be better to do it now. WDYT?

Reply:
should be in its own follow up PR IMHO